src/cmo.adoc

```diff
@@ -931,10 +931,13 @@ Encoding::
 Description::
 
 A *cbo.flush* instruction performs a flush operation on the cache block whose
-effective address is the base address specified in _rs1_. The offset operand may
-be omitted; otherwise, any expression that computes the offset shall evaluate to
-zero. The instruction operates on the set of coherent caches accessed by the
-agent executing the instruction.
+that contains the address specified in _rs1_. It is not required that _rs1_ is
+aligned to the size of a cache block. On faults the address stored in the *tval
+CSR is _rs1_; not the base address of the cache block. The instruction operates
+on the set of coherent caches accessed by the agent executing the instruction.
```

Comment:
A CMO to flush/invalidate/clean should take effect even if the cache block is cached in a coherent cache that is not directly accessible to the agent executing the corresponding CMO. For instance, the cache block may be cached in the private cache of a hart should be flushed/invalidated/cleaned if a CMO addressing that cache block was executed by another hart.

Reply:
It does say this elsewhere:

> Additionally, if a coherent agent in the set executes a CBO instruction that specifies the cache block, the resulting operation shall apply to any and all of the copies in the caches that can be accessed by the load and store operations from the coherent agents.

So I think it's reasonably clear, but if you want to clarify it more here I think it can go in a separate PR.

Reply:
"that can be accessed by the load and store operations" - it could be interpreted that CMO will not operate on private caches of other harts because they cannot be "accessed" by load/store operations performed on the hart that executes the CMO. The other interpretation would be that "accessed" includes forwarding of data from private caches of remote harts to the private cache of the hart executing the load/store. I would like to hear @dkruckemyer-ventana interpretation.

Reply:
Yeah I think that's just an oversight here. The other bit I quoted is much more explicit and clearly agrees with what you said.

Reply:
Please rewrite "On faults the address stored in the *tval CSR is _rs1_; not the base address of the cache block." as "On faults, the faulting virtual address is considered to be the value in _rs1_, rather than the base address of the cache block."

This improves separation of concerns, because faulting virtual addresses might end up in places other than *tval.